### ISSUE TITLE
MapView component is missing the possibility to forward ref and doesn't seem to work with useMap-hook

### ISSUE THREAD

**Issue by Defrosted** [aws-amplify/amplify-ui#1790]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

Create React App

### Please describe your bug.

The MapView wrapper component doesn't take a ref parameter at all so that it would be possible to access the Map API of the underlying Map-component from the react-map-gl package (as described in the [documentation](https://visgl.github.io/react-map-gl/docs/api-reference/map) for the Map component). The useMap hook provided by react-map-gl doesn't seem to work with the MapView component either, forcing me to use the pure JS maplibre-gl-js-amplify package instead of React components. I didn't delve a lot into the useMap-hook option, since it would require me to do extensive refactoring.

### What's the expected behaviour?

The expected behaviour would be to be able to pass refs and use the useMap-hook as documented by [react-map-gl](https://visgl.github.io/react-map-gl/docs/api-reference/map) documentation the MapView is claiming to be fully compatible with.

Example for original Map component from the react-map-gl documentation
```typescript
import * as React from 'react';
import Map from 'react-map-gl';

function App() {
  const mapRef = React.useRef();

  const onMapLoad = React.useCallback(() => {
    mapRef.current.on('move', () => {
      // do something
    });
  }, []);

  return <Map ref={mapRef} onLoad={onMapLoad} />;
}
```

### Help us reproduce the bug!

Trying to pass a ref to the MapView component isn't supported and is caught by Typescript.

```typescript
import { MapView } from '@aws-amplify/ui-react'
import { useRef } from 'react';
import { MapRef } from 'react-map-gl';
const Component: React.FC = () => {
  const mapRef = useRef<MapRef | undefined>(undefined);
  return (
    <MapView ref={mapRef} />
  );
}
```

### Additional information and screenshots

@aws-amplify/ui-react version 2.15.6
maplibre-gl version 1.15.3
react-map-gl version 7.0.11

**Comment by Defrosted** [comment]
Hi @slaymance. Big thanks for the quick reaction and work on the issue! Being able to pass the ref would be great, as currently I'm limited to using pure JS via maplibre-gl-js-amplify which is not working as desired (and gives errors) if there's an update to the map in between component renders.

What kind of a schedule do you have regarding releases after the PR is merged?

**Comment by imraymondlee** [comment]
Hi, we are also looking to be able to use `ref` from `MapView` component so we can use map methods from MapLibre.

**Comment by dreamorosi** [comment]
Came here to open the issue and found one already open.

The underlying `Map` component from `react-map-gl` accepts a ref of type `MapRef` (example).

Having this ref is needed to access the map component which in turn is essential to perform actions on the map like `flyTo` or simply `getBounds`:

```ts
mapRef.current?.flyTo({center: [longitude, latitude], duration: 2000});
```

**Comment by slaymance** [comment]
Ref forwarding has been included in the [v2.17.0](https://github.com/aws-amplify/amplify-ui/releases/tag/%40aws-amplify%2Fui-react%402.17.0) release.
